README.md

```diff
@@ -153,34 +156,90 @@ Loading...
 </section>
 
 <!-- later during the loading process, e.g. when the gallery content is ready -->
-<template for="photo-gallery" mode="replace">
+<template for="photo-gallery">
   ... the actual gallery...
 </template>
 ```
 
-#### Details of `<template for=... mode...>`
+#### Details of `<template for=...>`
+
+1. The `for` attribute behaves like `<label for>`, pointing to an ID.
+1. When a `<template for>` is discovered, its contents are streamed into its target position.
+1. The template itself is just a DOM directive. It doesn't stay in the DOM after being processed.
+1. Potentially, we can make this template point to a *position* in the DOM rather than an ID, e.g. one or two CSS selectors and replacing everything between them.
+   If we do that, the template does nothing and is discarded if it doesn't have anywhere to stream into.
 
-1. The `for` attribute behaves like `<label for>`, pointing to an ID
-1. `mode` can be `replace`, `append` or `prepend`, and be extended in the future to include things like `merge`, `loading` and `error`.
-1. In general, the `<template>` itself does not stay in the DOM, kind of like `<template shadowrootmode>`. However the details of this are TBD.
-1. When a `<template for>` is discovered, its contents are streamed into its target element.
- 
 This is, in essence, a proposed solution for https://github.com/whatwg/html/issues/2791. 
 
-### Part 2: Composable app fragments (views)
+### Part 2: URL matching with views & routes
+
+To effectively match between parts of the document and the state of the app, there is commonly a relationship between the document's URL and active/visible components of the app.
+This is commonly referred to in frameworks as "routes".
 
-In addition to being able to declaratively stream content into the document out-of-order, we want to enable a declarative relationship between a fragment of an application and the document's URL.
-Straw man proposal is to do this with a new element called `<view>`, with a `match` attribute that accepts a `URLPattern`, and intercepts matching navigations.
+The proposal here is to make routes a first-class citizen in HTML, and using that (at first) for two things:
+1. Wrapping certain parts of the documents that belong to a certain route with a `<view>` element that describes that relationship.
+2. CSS at-rule conditions that get activated by route-matching, similar to media-queries:
 
 ```html
-<view match="/articles/:article-id" id=main-article>
+<!-- Similar shape to importmap -->
+<script type=routemap>
+  {
+    "movies": { "pattern": { "pathname": "/movies/:movie_id" } },
+    "about":  { "pattern": "/about" },
+    "home":   { "pattern": ["/home", "/"] }
+  ]
+</script>
+
+<style>
+@route home {
+  nav {
+    display: none;
+  }
+}
+</style>
+
+<body>
+   <view match=home>
+   </view>
+   <view match=about>
+   </view>
+   <view match=movies>
+   </view>
+</body>
+```
+
+#### Details of `<view match=...>` and `<script type=routemap>`
```

Comment:
Bikeshedding: since this now uses `routemap` instead of raw URL patterns, perhaps the `match` attribute should be something like `route` (or `matchroute` or `whenroute`)?

That would also be more clear if this were added as a free-standing attribute instead of as part of a new `<view>` element.